### DISCUSSION TITLE
why this project is not thread safe

### DISCUSSION THREAD

**Discussion by yujie1007** [cesanta/mongoose#1451] [Ask questions, get answers] [answered]
when I use  mg_ws_conn and mg_mgr_poll on different threads，my application maybe  crash。

**Comment by cpq** [comment, marked as answer]
It is not thread safe intentionally by design.
It is explicitly noted in the documentation that is it NOT thread safe, see a red note in the https://mongoose.ws/documentation/#2-minute-integration-guide

**Reply by yujie1007** [reply]
thank you for your answer
